src/Dapper.FluentMap/Utils/ReflectionHelper.cs

```diff
@@ -31,11 +33,22 @@ public static MemberInfo GetMemberInfo(LambdaExpression lambda)
                     case ExpressionType.MemberAccess:
                         var memberExpression = (MemberExpression)expr;
                         var baseMember = memberExpression.Member;
+                        Type paramType;
+
+                        while (!(memberExpression is null))
+                        {
+                            paramType = memberExpression.Type;
+                            if (paramType.GetMembers().Any(member => member.Name == baseMember.Name))
+                            {
+                                return paramType.GetMember(baseMember.Name).FirstOrDefault();
+                            }
+
+                            memberExpression = memberExpression.Expression as MemberExpression;
+                        }
 
                         // Make sure we get the property from the derived type.
-                        var paramType = lambda.Parameters[0].Type;
-                        var memberInfo = paramType.GetMember(baseMember.Name)[0];
-                        return memberInfo;
+                        paramType = lambda.Parameters.FirstOrDefault().Type;
```

Comment:
When reading the code `FirstOrDefault` tells you that there might not be a value present or that the collection is not an array/list. Beside that, I tend to avoid overhead from LINQ as much as possible.